lib/events.js

```diff
@@ -351,6 +351,54 @@ EventEmitter.prototype.prependOnceListener =
       return this;
     };
 
+// An awaitable one-time event listener. Registers a once handler and
+// returns an associated promise.
+EventEmitter.prototype.when =
+    function when(type, options = { prepend: false }) {
+      if (options == null || typeof options !== 'object') {
+        // This must be lazy evaluated
+        const errors = require('internal/errors');
+        throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
+                                   'options', 'object');
+      }
+      // Do not pull these in at the top, they have to be lazy evaluated
+      const {
```

Comment:
Is this the best/fastest way to pull these in (calling `process.binding()` every time)? Why about making these top-level variables and adding a check if one of them is `undefined` then set them from `process.binding('util')`?